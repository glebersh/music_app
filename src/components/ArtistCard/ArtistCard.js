import React from 'react';
import './ArtistCard.css';
import { Link } from 'react-router-dom';
import { Image, Flex, Tag, Text, Box } from '@chakra-ui/react';
import { useSelector } from 'react-redux';


const ArtistCard = ({ name, imgURL, artistID }) => {
  const link = `artists/${artistID}`;
  const isLoading = useSelector(state => state.searchReducer.isLoading);
  return (
    <div data-testid='artist-card'>
      {isLoading ?
        <Flex direction='column' gap='.5em' justify='flex-start' mt='3em'
          align='flex-start'>

          <Image src={imgURL} w='240px' h='240px' loading="lazy"
            fallbackSrc='https://via.placeholder.com/240' />

          <Link to={link} role='link'>
            <Text fontSize='1em' transtion='color 0.33s'
              _hover={{ color: 'primary' }}
              w='100%'>{name}
            </Text>
          </Link >
          <Tag w='100px' fontSize='1em'
            paddingInlineStart='30px'>Artist</Tag>
        </Flex > : null}
    </div>
  )
};

export default ArtistCard;

