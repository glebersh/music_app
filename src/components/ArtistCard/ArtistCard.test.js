import ArtistCard from "./ArtistCard";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import userEvent from "@testing-library/user-event";

import * as reduxHooks from 'react-redux';
import { BrowserRouter } from "react-router-dom";
jest.mock('react-redux');
const mockedUseSelector = jest.spyOn(reduxHooks, 'useSelector');



describe('Artist Card test', () => {
  it('Should render and match snapshot when data loaded', () => {

    mockedUseSelector.mockReturnValue(true);
    const card = render(
      <BrowserRouter>
        <ArtistCard
          name='Red Velvet'
          imgURL='https://via.placeholder.com/300.png'
          artistID='1z4g3DjTBBZKhvAroFlhOM'
        />
      </BrowserRouter>);
    expect(card).toMatchSnapshot();
  });

  it('Should navigate to artist page on click', () => {

    mockedUseSelector.mockReturnValue(true);
    const card = render(
      <BrowserRouter>
        <ArtistCard
          name='Red Velvet'
          imgURL='https://via.placeholder.com/300.png'
          artistID='1z4g3DjTBBZKhvAroFlhOM'
        />
      </BrowserRouter>);
    expect(card.getByRole('link').outerHTML).toContain('artists/1z4g3DjTBBZKhvAroFlhOM');
  });

  it('Should not render if data not loaded', () => {

    mockedUseSelector.mockReturnValue('');
    const card = render(
      <BrowserRouter>
        <ArtistCard
          name='Red Velvet'
          imgURL='https://via.placeholder.com/300.png'
          artistID='1z4g3DjTBBZKhvAroFlhOM'
        />
      </BrowserRouter>);
    expect(card.getByTestId('artist-card')).toBeEmptyDOMElement();
  });
});