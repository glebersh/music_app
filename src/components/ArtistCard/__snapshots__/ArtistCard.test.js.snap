// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Card test Should render and match snapshot when data loaded 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="artist-card"
      >
        <div
          class="css-1kngx9c"
        >
          <img
            class="chakra-image css-1dtlibk"
            loading="lazy"
            src="https://via.placeholder.com/300.png"
          />
          <a
            href="/artists/1z4g3DjTBBZKhvAroFlhOM"
            role="link"
          >
            <p
              class="chakra-text css-fp4dd8"
              transtion="color 0.33s"
            >
              Red Velvet
            </p>
          </a>
          <span
            class="css-14lud7x"
          >
            Artist
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="artist-card"
    >
      <div
        class="css-1kngx9c"
      >
        <img
          class="chakra-image css-1dtlibk"
          loading="lazy"
          src="https://via.placeholder.com/300.png"
        />
        <a
          href="/artists/1z4g3DjTBBZKhvAroFlhOM"
          role="link"
        >
          <p
            class="chakra-text css-fp4dd8"
            transtion="color 0.33s"
          >
            Red Velvet
          </p>
        </a>
        <span
          class="css-14lud7x"
        >
          Artist
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
